package opensolid;

message ObjectMessage {
    repeated KeyValueMessage key_value_pair = 1;
}

message KeyValueMessage {
    required KeyMessage key = 1;
    required ValueMessage value = 2;
}

message KeyMessage {
    optional int32 int_key = 1;
    optional string string_key = 2;
}

message ValueMessage {
    optional int32 int_value = 1;
    optional double double_value = 2;
    optional string string_value = 3;
    optional ObjectMessage object_value = 4;
}
